src/plots/cartesian/axis_defaults.js

```diff
@@ -142,6 +138,11 @@ module.exports = function handleAxisDefaults(containerIn, containerOut, coerce,
         delete containerOut.zerolinewidth;
     }
 
+    // fill in categories
+    containerOut._initialCategories = axType === 'category' ?
+        orderedCategories(letter, containerOut.categoryorder, containerOut.categoryarray, options.data) :
+        [];
+
     return containerOut;
```

Comment:
@etpinard a quick chat or pointer about containerIn vs Out would be interesting as I kept using the former.